flowkit_ui_backend/main.py

```diff
@@ -101,15 +101,18 @@ async def permission_exception_handler(request: Request, e: PermissionError):
 
 
 @app.exception_handler(StarletteHTTPException)
-async def http_exception_handler(request, e):
+async def http_exception_handler(request, e:StarletteHTTPException):
     logger.error(f"{type(e)}: {str(e)}")
-    return await http_exception_handler(request, e)
+    if e.status_code == HTTPStatus.UNAUTHORIZED:
```

Comment:
I'm a touch concerned I'm hitting Chesterton's Fence here - I don't know why these functions were recursing, and it was causing recursion depth errors in my local dev environment but there might be some deployment subtlety here that I'm not catching.